RFS/src/main/java/com/rfs/common/LuceneDocumentsReader.java

```diff
@@ -11,40 +10,65 @@
 import org.apache.lucene.index.IndexReader;
 import org.apache.lucene.store.FSDirectory;
 import org.apache.lucene.util.BytesRef;
+import reactor.core.publisher.Flux;
 
 public class LuceneDocumentsReader {
     private static final Logger logger = LogManager.getLogger(LuceneDocumentsReader.class);
 
-    public static List<Document> readDocuments(Path luceneFilesBasePath, String indexName, int shardId) throws Exception {
+    public Flux<Document> readDocuments(Path luceneFilesBasePath, String indexName, int shardId) {
```

Comment:
Why does your LuceneDocumentReader now take a hard dependency on your HTTP client library?
It might be better to make this a collection or stream & then adapt later so that you can switch client implementations out.